### PR TITLE
[FWCore][clang-tidy] make deleted function public

### PR DESCRIPTION
**Pull request by smuzaffar**
We have over 25K clang-tidy warnings about `deleted member function should be public [modernize-use-equals-delete]`. 9K of these come from FWCore sub-system.
Locally tested, it build and all of the `modernize-use-equals-delete` warnings are fixed.

[a]
```
FWCore/Framework/interface/EventSetupImpl.h:86:5: warning: deleted member function should be public [modernize-use-equals-delete]
    EventSetupImpl() = delete;
    ^
FWCore/Framework/interface/one/EDAnalyzer.h:57:7: warning: deleted member function should be public [modernize-use-equals-delete]
      EDAnalyzer(const EDAnalyzer&) = delete;
      ^
FWCore/Framework/interface/one/EDAnalyzer.h:58:25: warning: deleted member function should be public [modernize-use-equals-delete]
      const EDAnalyzer& operator=(const EDAnalyzer&) = delete;
```

### CODE REVIEW

**Comment by Dr15Jones** [comment]
Please test

File: FWCore/FWLite/interface/AutoLibraryLoader.h [hunk 1]
```diff
@@ -24,6 +24,8 @@ class AutoLibraryLoader {
 private:
   static bool enabled_;
   AutoLibraryLoader();
+
+public:
   AutoLibraryLoader(const AutoLibraryLoader&) = delete;                   // stop default
   const AutoLibraryLoader& operator=(const AutoLibraryLoader&) = delete;  // stop default
```
Comment (makortel) [inline FWCore/FWLite/interface/AutoLibraryLoader.h:30]: Not a big deal in practice, but I'd prefer these to be moved to the existing `public:` section above. (that would also be more according to our code rules)

File: FWCore/MessageService/src/ELdestination.h [hunk 2]
```diff
@@ -134,7 +134,7 @@ namespace edm {
 
       // -----  Verboten methods:
       //
-    private:
+    public:
       ELdestination(const ELdestination& orig) = delete;
       ELdestination& operator=(const ELdestination& orig) = delete;
```
Comment (makortel) [inline FWCore/MessageService/src/ELdestination.h:139]: I'd prefer to move these to the top in the existing `public:` section.

File: FWCore/MessageService/src/ELoutput.h [hunk 3]
```diff
@@ -98,6 +98,7 @@ namespace edm {
       bool wantTimestamp, wantModule, wantSubroutine, wantText, wantSomeContext, wantSerial, wantFullContext,
           wantTimeSeparate, wantEpilogueSeparate, preambleMode;
 
+    public:
       // --- Verboten method:
       //
       ELoutput& operator=(const ELoutput& orig) = delete;
```
Comment (makortel) [inline FWCore/MessageService/src/ELoutput.h:104]: I'd prefer to move this to the top in the existing `public:` section.

Comment (smuzaffar) [reply FWCore/MessageService/src/ELoutput.h:104]: sure I will take care of these. I just wanted to make minimal changes :-) I will update the PR to move these deleted constructor to top

**Comment by makortel** [comment]
@cmsbuild, please test

**Comment by makortel** [comment]
+1

The test failed in the reference IB.

**Comment by perrotta** [comment]
+1
- Technical, no changes in outputs

**Comment by perrotta** [comment]
+1
- Technical, no changes in outputs

**Comment by perrotta** [comment]
merge

**Pull request merged**